optaplanner-wb-screens/optaplanner-wb-domain-editor/optaplanner-wb-domain-editor-backend/src/test/java/org/optaplanner/workbench/screens/domaineditor/backend/server/PlannerDataModelerEventObserverTest.java

```diff
@@ -0,0 +1,140 @@
+    }
+
+    @Test
+    public void onDataObjectDeletedDrlFileDoesntExist() {
+        testDrlFileDeletedAndCreated( false, false );
+    }
+
+    private void testDrlFileDeletedAndCreated( boolean createDrlFile, boolean drlFileExists ) {
+        Path resourcesPath = mock( Path.class );
+        when( resourcesPath.toURI() ).thenReturn( "default://PlannerDataModelerEventObserverTest/src/main/resources/foo" );
+
+        org.guvnor.common.services.project.model.Package resourcesPackage = mock( Package.class );
+        when( resourcesPackage.getPackageMainResourcesPath() ).thenReturn( resourcesPath );
+        when( kieProjectService.resolvePackage( any() ) ).thenReturn( resourcesPackage );
+
+        when( ioService.exists( any() ) ).thenReturn( drlFileExists );
+
+        DataObject dataObject = mock( DataObject.class );
+        when( dataObject.getAnnotation( PlanningSolution.class.getName() ) ).thenReturn( mock( Annotation.class ) );
+        when( dataObject.getSuperClassName() ).thenReturn( "org.optaplanner.core.impl.domain.solution.AbstractSolution<" + SimpleScore.class.getName() + ">" );
+
+        Path dataObjectPath = mock( Path.class );
+        when( dataObjectPath.toURI() ).thenReturn( "default://PlannerDataModelerEventObserverTest/src/main/foo/DataObject.java" );
+
+        if ( createDrlFile ) {
+            DataObjectSavedEvent event = (DataObjectSavedEvent) new DataObjectSavedEvent().withCurrentDataObject( dataObject ).withPath( dataObjectPath );
+            observer.onDataObjectSaved( event );
+        } else {
+            DataObjectDeletedEvent event = (DataObjectDeletedEvent) new DataObjectDeletedEvent().withCurrentDataObject( dataObject ).withPath( dataObjectPath );
+            observer.onDataObjectDeleted( event );
+        }
+
+        if ( drlFileExists ) {
+            Path drlFilePath = PathFactory.newPath( "scoreHolderGlobalDefinition.drl", "default://PlannerDataModelerEventObserverTest/src/main/resources/foo/scoreHolderGlobalDefinition.drl" );
+
+            verify( drlTextEditorService, times( 1 ) ).delete( drlFilePath, "Delete existing score holder definition." );
+        } else {
+            verify( drlTextEditorService, times( 0 ) ).delete( any( Path.class ), anyString() );
+        }
+
+        if ( createDrlFile ) {
+            verify( drlTextEditorService, times( 1 ) ).create( resourcesPath,
+                                                               "scoreHolderGlobalDefinition.drl",
+                                                               getScoreHolderDrlFileContent( SimpleScore.class ),
+                                                               "Create score holder definition for score type " + SimpleScore.class.getSimpleName() );
+        }
+    }
+
+    private String getScoreHolderDrlFileContent( Class<? extends Score> scoreType ) {
+        String lineSeparator = System.lineSeparator();
+        StringBuilder contentStringBuilder = new StringBuilder();
```

Comment:
Soft suggestion: there's no need to use a StringBuilder explicitly here, plain String concatenation isn't any slower because there are no loops, so the javac compiler will actually uses a StringBuilder for plain String concatenation (again, because there are no loops).